joomla/joomla-cms
# Add plugin events to com_contact (redo)

This is a redo of #3630, using the current staging branch and with some improvements.
### Contents

This PR implements the following plugin events for plugins (group: content):

| Event | Views |
| --- | --- |
| onContentPrepare | <ul><li>contact</li><li>category (for category title, category description and every contact)</li></ul> |
| onContentAfterTitle | <ul><li>contact</li><li>category (for every contact)</li></ul> |
| onContentBeforeDisplay | <ul><li>contact</li><li>category (for every contact)</li></ul> |
| onContentAfterDisplay | <ul><li>contact</li><li>category (for every contact)</li></ul> |
### Testing instructions
1. Install the patch.
2. If you don't have any plugin installed that targets com_contact, nothing should change in any view (especially in the com_contact views).
3. Install a content plugin that uses the events mentioned above in com_contact.
   You can either write a new one, modify one of the existing content plugins, or use [this small plugin for testing purposes](https://github.com/Harmageddon/plg_contact_events).
4. Make sure that the plugin works in the desired way.
5. Please test Beez (default and encyclopedia layout), too!
